File: check_braces.js

const fs = require('fs');

const content = fs.readFileSync('index.html', 'utf8');
const lines = content.split('\n');

let inScript = false;
let balance = 0;
let stack = [];

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];

    if (line.includes('<script>')) {
        inScript = true;
        console.log(`Script starts at line ${i + 1}`);
        continue;
    }
    if (line.includes('</script>')) {
        inScript = false;
        console.log(`Script ends at line ${i + 1}, balance: ${balance}`);
    }

    if (!inScript) continue;

    // Simple char loop with state for strings
    let state = 'code'; // code, string_double, string_single, template

    for (let j = 0; j < line.length; j++) {
        const char = line[j];

        if (state === 'code') {
            if (char === '"') state = 'string_double';
            else if (char === "'") state = 'string_single';
            else if (char === '`') state = 'template';
            else if (char === '{') {
                balance++;
                stack.push(i + 1);
            } else if (char === '}') {
                balance--;
                if (balance < 0) {
                    console.log(`Error: Unexpected } at line ${i + 1} col ${j + 1}`);
                    process.exit(1);
                }
                stack.pop();
            } else if (char === '/' && line[j + 1] === '/') {
                break; // comment
            }
        } else if (state === 'string_double') {
            if (char === '"' && line[j - 1] !== '\\') state = 'code';
        } else if (state === 'string_single') {
            if (char === "'" && line[j - 1] !== '\\') state = 'code';
        } else if (state === 'template') {
            if (char === '`' && line[j - 1] !== '\\') state = 'code';
        }
    }
}

if (balance > 0) {
    console.log(`Error: Missing } at EOF. Balance is ${balance}. Last open brace was at line ${stack[stack.length - 1]}`);
} else {
    console.log("Braces seem balanced.");
}

File: check_braces.py

import re

with open(r'c:\Users\NewTech\.gemini\antigravity\scratch\oudelle-vercel-complete\index.html', 'r', encoding='utf-8') as f:
    lines = f.readlines()

in_script = False
balance = 0
last_open_line = -1

for i, line in enumerate(lines, 1):
    if '<script>' in line:
        in_script = True
        print(f"Script starts at {i}")
        continue
    if '</script>' in line:
        in_script = False
        print(f"Script ends at {i}, balance: {balance}")
        if balance != 0:
             print(f"Unbalanced at end of script! Balance: {balance}")
        continue
    
    if notInScript: continue

    line_stripped = line.strip()
    # Simplified parser: count braces outside strings/comments?
    # This is hard to do correctly with simple counters if there are complex strings.
    # But let's try basic counting for now, as it's likely a simple block issue.
    
    # Remove strings to avoid counting braces inside them
    # This is a naive removal, but often sufficient for debugging
    line_no_str = re.sub(r'([\"\']).*?\1', '', line) 
    # Remove template strings (backticks) - very naive, doesn't handle multiline
    # JS template strings can be multiline, so line-by-line is risky.
    
    for char in line_no_str:
        if char == '{':
            balance += 1
            last_open_line = i
        elif char == '}':
            balance -= 1

    if balance < 0:
        print(f"Error: Negative balance at line {i}")
        break

if balance > 0:
    print(f"Error: Positive balance {balance} at EOF. Last open brace around line {last_open_line}")
